src/Titon/Context/Depository.hh

```diff
@@ -166,6 +221,10 @@ class Depository {
             return $this->getRegisteredItem($alias, ...$arguments);
         }
 
+        if (is_string($alias) && $this->isInServiceProvider($alias)) {
```

Comment:
Since we are doing like 3 `is_string()` checks, lets reduce it to 1 and have nested ifs.